canopeum_frontend/src/pages/Login.tsx

```diff
@@ -1,150 +1,116 @@
-
-  const [loginError, setLoginError] = useState<string | undefined>()
-
-  const validateEmail = () => {
-    if (!email) {
-      setEmailError('required')
-
-      return false
-    }
-
-    setEmailError(undefined)
-
-    return true
-  }
-
-  const validatePassword = () => {
-    if (!password) {
-      setPasswordError('required')
-
-      return false
-    }
-
-    setPasswordError(undefined)
-
-    return true
-  }
-
-  const validateForm = () => {
-    // Do not return directly the method calls;
-    // we need each of them to be called before returning the result
-    const emailValid = validateEmail()
-    const passwordValid = validatePassword()
-
-    return emailValid
-      && passwordValid
-  }
+type LoginFormInputs = {
+  email: string,
+  password: string,
+  rememberMe: boolean,
+}
 
-  const onLoginClick = async () => {
-    const isFormValid = validateForm()
-    if (!isFormValid) return
+const Login = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, touchedFields },
+  } = useForm<LoginFormInputs>({ mode: 'onTouched' })
 
+  const onSubmit: SubmitHandler<LoginFormInputs> = async formData => {
     try {
       const response = await getApiClient().authenticationClient.login(
         new LoginUser({
-          email: email.trim(),
-          password,
+          email: formData.email.trim(),
+          password: formData.password,
         }),
       )
 
-      storeToken(response.token, rememberMe)
+      storeToken(response.token, formData.rememberMe)
       authenticate(response.user)
     } catch {
       setLoginError(translate('auth.log-in-error'))
     }
   }
 
+  const { authenticate } = useContext(AuthenticationContext)
+  const { t: translate } = useTranslation()
+  const { getApiClient } = useApiClient()
+
+  const [loginError, setLoginError] = useState<string | undefined>()
+
   return (
     <AuthPageLayout>
       <div style={{ flexGrow: '0.4', display: 'flex', alignItems: 'center' }}>
         <h1 style={{ textAlign: 'center' }}>{translate('auth.log-in-header-text')}</h1>
       </div>
 
-      <div className='col-10 col-sm-6 col-md-8 col-xl-6 d-flex flex-column gap-4'>
+      <form
+        className='col-10 col-sm-8 col-xl-6 d-flex flex-column gap-4'
+        onSubmit={handleSubmit(onSubmit)}
+      >
         <div className='w-100'>
-          <label htmlFor='email-input'>{translate('auth.email-label')}</label>
+          <label htmlFor='email'>{translate('auth.email-label')}</label>
           <input
-            aria-describedby='email'
+            aria-describedby='emailHelp'
             className={`form-control ${
-              emailError
-                ? 'is-invalid'
+              touchedFields.email && errors.email
+                ? formClasses.invalidFieldClass
                 : ''
             }`}
-            id='email-input'
-            onBlur={() => validateEmail()}
-            onChange={event => setEmail(event.target.value)}
-            type='email'
```

Comment:
See above comment for the id linked to the label

You should keep at the very least the name or type to email. type is even better.

This will tell screen readers (including form autocompletion browser plugins and mobile apps) that this is an email field. It'll also update the keyboard on mobile device to their email mode (where symbols like `@` are shown in quick selection)